clippy_lints/src/strings.rs

```diff
@@ -154,8 +155,8 @@ impl LintPass for StringLitAsBytes {
 }
 
 impl EarlyLintPass for StringLitAsBytes {
-    fn check_expr(&mut self, cx: &EarlyContext<'_>, e: &crate::syntax::ast::Expr) {
-        use crate::syntax::ast::{ExprKind, LitKind};
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, e: &ast::Expr) {
+        use self::ast::{ExprKind, LitKind};
```

Comment:
Why does this need the `self::`... path segment? is that what this is called? The type for `e` in the function args doesn't need it.

Reply:
The type in the function call just looks for the `use` statements of the file and can use them directly. `use`-statements on the other hand also need to be able to import stuff from outside the current file, so you have to specify where it should start looking. 
- Crate level: `crate::foo` in 2018 edition, `::foo` in 2015 edition
- Module level: `self::foo`
- Parent-module level: `super::foo` (parent-parent-module: `super::super::foo`, and so on)

If you just write `use foo` in the 2018 edition (`extern crate` got removed) it will look for a crate named `foo`, whereas `use self::foo` looks for a module/function/... inside the current module.

[Playground](https://play.rust-lang.org/?gist=2e2088dd2f9ef0ed61c610962b20eb3b&version=beta&mode=debug&edition=2018)